core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -340,4 +351,25 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
     sc.conf.getTimeAsSeconds("spark.mesos.rejectOfferDurationForUnmetConstraints", "120s")
   }
 
+  protected def getExecutorRelatedPort(sc: SparkContext, configProperty: String): Int = {
+    // either: spark.blockManager.port or spark.executor.port
+    sc.conf.getInt(s"$configProperty", 0)
+  }
+
+  protected def checkPorts(sc: SparkContext, ports: List[(Long, Long)]): Boolean = {
+
+    def checkIfInRange(port: Int, ps: List[(Long, Long)]): Boolean = {
+      ps.takeWhile(r => !(r._1 <= port && r._2 >= port)).size != ps.size
+    }
+
+    val portsToCheck = List(getExecutorRelatedPort(sc, "spark.executor.port"),
+      getExecutorRelatedPort(sc, "spark.blockManager.port"))
+    val (_, nonZeroPorts) = portsToCheck.partition(_ == 0)
```

Comment:
I'd go for the simpler version: `val nonZeroPorts = portsToCheck.filter(_ != 0)`.